### PR TITLE
UpgradeTest stability improvements

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Bugfix

### Description

Improvement to the stability of the UpgradeTests.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@kornys I've been investigating the root cause of some upgrade test instability.   This patch addresses one source of instability - the failure of older EnMasse installs to come ready prior to the upgrade commencing. We see this failure very frequently so I think it is worth the workaround.

I also want to leave the capturing of deployments/statefulsets in GLC.  This is part of a separate investigation into a different upgrade issue which happens less frequently.

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4 testcase=UpgradeTest

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=upgrade ocp4

File: systemtests/src/main/java/io/enmasse/systemtest/logs/GlobalLogCollector.java [hunk 1]
```diff
@@ -346,6 +346,10 @@ public static void saveInfraState(Path path, String infraNamespace, ThrowableRun
             Files.writeString(path.resolve("events.txt"), KubeCMDClient.getAllEvents().getStdOut());
             Files.writeString(path.resolve("describe_nodes.txt"), KubeCMDClient.describeNodes().getStdOut());
 
+            // Investigating sporadic upgrade issue.
+            Files.writeString(path.resolve("deployments.yml"), KubeCMDClient.runOnClusterWithoutLogger("get", "deployments", "-o", "yaml").getStdOut());
```
Comment (kornys) [inline systemtests/src/main/java/io/enmasse/systemtest/logs/GlobalLogCollector.java:350]: @k-wall can you move it please into loop with all systemtests namespaces? I think it should be useful to collect these info for all systemtests namespaces. (loop is in same file line 321...)

Comment (k-wall) [reply systemtests/src/main/java/io/enmasse/systemtest/logs/GlobalLogCollector.java]: Addressed.

**Pull request merged**